cfme/common/provider.py

```diff
@@ -100,6 +100,10 @@ def type(self):
         return self.type_name
 
     @property
+    def id(self):
```

Comment:
**REQUIRED:** Two things here
1) The ```get_provider_id()``` method used requires a ```provider_name``` which you would probably ask why? I agree it shouldn't need a provider name if it is a class method.
2) The ```get_provider_id()``` method isn't used at all in the codebase. So I propose we switch the code into the ```id``` property and drop the requirement for the arg.  It's a required arg to, so any calls to this I would assume would fail.

Reply:
I verified it does fail....

Reply:
@psav, Done

Reply:
If we have this implemented above using REST, why do we need to do it again here via the UI?

Reply:
@psav, I'm not quite sure why we don't have an id property in BaseProvider class in common/provider.py  in the first place, but have one in the InfraProvider class cfme/infrastructure/provider/__init__.py.Do let me know if I'm missing something.Anyway, I placed the id property in  the BaseProvider class in common/provider.py and removed it from cfme/infrastructure/provider/__init__.py since that would be just a duplicate override.Both InfraProvider and CloudProvider are child classes of CloudInfraProvider, which in turn is a child class of  BaseProvider class.I tested the code for both cloud and infra providers and it works fine.

Also, that new id property in BaseProvider class uses REST to fetch provider ID.If you are okay with this, I could push my code changes.

Reply:
@psav, I've made changes as per my previous comment.I've also updated the PRT results in the comment section.

Could you review and merge the PR?